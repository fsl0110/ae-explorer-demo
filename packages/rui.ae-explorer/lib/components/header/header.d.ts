/// <reference types="@emotion/core" />
import React from "react";
import { RouteComponentProps } from "react-router-dom";
import { Data } from "../../types";
export interface Props extends RouteComponentProps<{
    id: string;
}> {
    data: Data[];
}
export declare const Header: import("react").ComponentClass<Pick<Props, "data">, any> & import("react-router").WithRouterStatics<React.FunctionComponent<Props>>;
