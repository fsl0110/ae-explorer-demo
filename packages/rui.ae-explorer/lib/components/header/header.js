"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var react_1 = tslib_1.__importDefault(require("react"));
var react_router_dom_1 = require("react-router-dom");
var rui_component_lib_1 = require("@dedrr/rui.component-lib");
var detailsTableManager_redux_1 = require("../../connections/detailsTableManager/detailsTableManager.redux");
var storeManager_redux_1 = require("../../connections/storeManager/storeManager.redux");
var groupsFilterManager_redux_1 = require("../../connections/groupsFilterManager/groupsFilterManager.redux");
var groupsTableManager_redux_1 = require("../../connections/groupsTableManager/groupsTableManager.redux");
var connections_1 = require("../../connections");
var help_1 = require("../help/help");
var HeaderWithRouter = function (_a) {
    var data = _a.data, location = _a.location;
    var arr = location.pathname.split("/");
    return (react_1.default.createElement(rui_component_lib_1.PageHeader, null,
        react_1.default.createElement(react_router_dom_1.Link, { to: "/" + arr[1] },
            react_1.default.createElement(rui_component_lib_1.BackButton, { tooltipTitle: "back", disabled: !arr[2] })),
        react_1.default.createElement(storeManager_redux_1.StoreManagerConnection, null),
        react_1.default.createElement("div", { className: "manager" },
            react_1.default.createElement(rui_component_lib_1.ModalButton, { iconName: "filter", tooltipTitle: "Data Filters Manager", disabled: !!arr[2] },
                react_1.default.createElement(groupsFilterManager_redux_1.GroupsFilterManagerConnection, null)),
            react_1.default.createElement(rui_component_lib_1.ModalButton, { iconName: "table", tooltipTitle: "Table Manager" }, !arr[2] ? react_1.default.createElement(groupsTableManager_redux_1.GroupsTableManagerConnection, null) : react_1.default.createElement(detailsTableManager_redux_1.DetailsTableManagerConnection, null))),
        react_1.default.createElement(rui_component_lib_1.ModalButton, { iconName: "question", tooltipTitle: "Help", width: 600 },
            react_1.default.createElement(help_1.Help, null)),
        react_1.default.createElement("div", null,
            react_1.default.createElement(connections_1.ShareConnection, null))));
};
exports.Header = react_router_dom_1.withRouter(HeaderWithRouter);
//# sourceMappingURL=header.js.map