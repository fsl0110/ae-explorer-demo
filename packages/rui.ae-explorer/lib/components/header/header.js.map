{"version":3,"file":"header.js","sourceRoot":"","sources":["../../../src/app/components/header/header.tsx"],"names":[],"mappings":";;;AAAA,wDAAgC;AAChC,qDAAuE;AACvE,8DAA6E;AAC7E,6GAA8G;AAC9G,wFAAyF;AACzF,6GAA8G;AAC9G,0GAA2G;AAC3G,iDAAkD;AAElD,qCAAkC;AAMlC,IAAM,gBAAgB,GAAc,UAAC,EAAgB;QAAf,cAAI,EAAE,sBAAQ;IAClD,IAAM,GAAG,GAAG,QAAQ,CAAC,QAAQ,CAAC,KAAK,CAAC,GAAG,CAAC,CAAC;IAEzC,OAAO,CACL,8BAAC,8BAAU;QACT,8BAAC,uBAAI,IAAC,EAAE,EAAE,MAAI,GAAG,CAAC,CAAC,CAAG;YACpB,8BAAC,8BAAU,IAAC,YAAY,EAAC,MAAM,EAAC,QAAQ,EAAE,CAAC,GAAG,CAAC,CAAC,CAAC,GAAI,CAChD;QACP,8BAAC,2CAAsB,OAAG;QAC1B,uCAAK,SAAS,EAAC,SAAS;YACtB,8BAAC,+BAAW,IAAC,QAAQ,EAAC,QAAQ,EAAC,YAAY,EAAC,sBAAsB,EAAC,QAAQ,EAAE,CAAC,CAAC,GAAG,CAAC,CAAC,CAAC;gBACnF,8BAAC,yDAA6B,OAAG,CACrB;YACd,8BAAC,+BAAW,IAAC,QAAQ,EAAC,OAAO,EAAC,YAAY,EAAC,eAAe,IACvD,CAAC,GAAG,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,8BAAC,uDAA4B,OAAG,CAAC,CAAC,CAAC,8BAAC,yDAA6B,OAAG,CACnE,CACV;QACN,8BAAC,+BAAW,IAAC,QAAQ,EAAC,UAAU,EAAC,YAAY,EAAC,MAAM,EAAC,KAAK,EAAE,GAAG;YAC7D,8BAAC,WAAI,OAAG,CACI;QACd;YACE,8BAAC,6BAAe,OAAG,CACf,CACK,CACd,CAAC;AACJ,CAAC,CAAC;AAEW,QAAA,MAAM,GAAG,6BAAU,CAAC,gBAAgB,CAAC,CAAC"}