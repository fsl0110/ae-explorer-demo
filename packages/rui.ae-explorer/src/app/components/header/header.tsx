import React, {FC} from "react";
import {Link, withRouter, RouteComponentProps} from "react-router-dom";
import {PageHeader, BackButton, ModalButton} from "@dedrr/rui.component-lib";
import {DetailsTableManagerConnection} from "../../connections/detailsTableManager/detailsTableManager.redux";
import {StoreManagerConnection} from "../../connections/storeManager/storeManager.redux";
import {GroupsFilterManagerConnection} from "../../connections/groupsFilterManager/groupsFilterManager.redux";
import {GroupsTableManagerConnection} from "../../connections/groupsTableManager/groupsTableManager.redux";
import {ShareConnection} from "../../connections";
import {Data} from "../../types";
import {Help} from "../help/help";

export interface Props extends RouteComponentProps<{id: string}> {
  data: Data[];
}

const HeaderWithRouter: FC<Props> = ({data, location}) => {
  const arr = location.pathname.split("/");

  return (
    <PageHeader>
      <Link to={`/${arr[1]}`}>
        <BackButton tooltipTitle="back" disabled={!arr[2]} />
      </Link>
      <StoreManagerConnection />
      <div className="manager">
        <ModalButton iconName="filter" tooltipTitle="Data Filters Manager" disabled={!!arr[2]}>
          <GroupsFilterManagerConnection />
        </ModalButton>
        <ModalButton iconName="table" tooltipTitle="Table Manager">
          {!arr[2] ? <GroupsTableManagerConnection /> : <DetailsTableManagerConnection />}
        </ModalButton>
      </div>
      <ModalButton iconName="question" tooltipTitle="Help" width={600}>
        <Help />
      </ModalButton>
      <div>
        <ShareConnection />
      </div>
    </PageHeader>
  );
};

export const Header = withRouter(HeaderWithRouter);
