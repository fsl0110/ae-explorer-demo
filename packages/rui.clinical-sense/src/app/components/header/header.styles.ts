import styled from "styled-components";

// basic exaple for styling html tags
export const HeaderStyles = styled.header`
  display: flex;
  align-items: center;
  background: #000;
  color: #fff;
  padding: 1rem;
  border-bottom: 1px solid #fff;
  height: 50px;

  h4 {
    color: #fff;
    margin: 0;
    margin-right: 1em;
    padding: 0;
  }

  .demo-text {
    padding-left: 1em;
  }
`;
