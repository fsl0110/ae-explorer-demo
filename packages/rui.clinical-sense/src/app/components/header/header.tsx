import React, {FC} from "react";
import {withRouter, RouteComponentProps} from "react-router-dom";
import {HeaderStyles} from "./header.styles";
import {Select} from "antd";

interface Props extends RouteComponentProps {
  handleChange: (payload: string) => void;
}

export const HeaderWithRouter: FC<Props> = ({handleChange, history}) => {
  const handleDataChange = (payload: string) => {
    handleChange(payload);
    history.push("/aeexplorer");
  };

  return (
    <HeaderStyles>
      <h4>Select Data</h4>
      <Select defaultValue="full" onChange={handleDataChange}>
        <Select.Option value="full">Full Data</Select.Option>
        <Select.Option value="data">Tiny Data</Select.Option>
      </Select>
      <h4 className="demo-text">(only for Demo)</h4>
    </HeaderStyles>
  );
};

export const Header = withRouter(HeaderWithRouter);
