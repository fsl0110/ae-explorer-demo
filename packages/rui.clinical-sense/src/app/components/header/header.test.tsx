import React from "react";
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import {Header} from "./header";

describe("<Header>", () => {
  it("Snapshot", () => {
    expect(toJson(shallow(<Header />))).toMatchSnapshot();
  });
});
