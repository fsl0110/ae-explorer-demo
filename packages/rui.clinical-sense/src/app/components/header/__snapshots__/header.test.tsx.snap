// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> Snapshot 1`] = `
<styled.header>
  <h1>
    Blueprint App
  </h1>
  <Link
    to="/"
  >
    Home
  </Link>
  <Link
    to="/contact"
  >
    Contact
  </Link>
</styled.header>
`;
